libbeat/autodiscover/autodiscover.go

```diff
@@ -178,7 +178,7 @@ func (a *Autodiscover) handleStop(event bus.Event) {
 
 		if runner := a.runners.Get(hash); runner != nil {
 			logp.Info("Autodiscover stopping runner: %s", runner)
-			runner.Stop()
+			go runner.Stop()
 			a.runners.Remove(hash)
```

Comment:
the hash should only be removed if stopping is finished I would think

Reply:
We actually want to remove it here, if a new event arrives to start the same module we want that to happen, even if the previous instance is still stopping

Reply:
I see, so adding and removing is synchronous. To make it more clear in the code we should add a comment here and move the stopping command after the removal of the hash.

Any concerns about leaking go routines here if `Stop()` gets stuck?